lib/ddtrace/contrib/kafka/events/produce_operation/send_messages.rb

```diff
@@ -15,6 +15,7 @@ module SendMessages
             def self.process(span, _event, _id, payload)
               super
 
+              span.set_tag(Ext::TAG_TOPIC, payload[:topic]) if payload.key?(:topic)
```

Comment:
For `send_messages` also, I think a buffer of messages that may cover multiple topics is being sent and therefore there is no topic in the instrumentation payload: 

https://github.com/zendesk/ruby-kafka/blob/93c45a8c531e8b447f1cd9127c8bc6ce8070c0e6/lib/kafka/produce_operation.rb#L53-L66

Reply:
@tjwp I see, thanks for pointing that out. 

It seems there may be other events emitted by ruby-kafka which are not processed in the ddtrace Kafka module (e.g. [ack_message.producer](https://github.com/zendesk/ruby-kafka/blob/93c45a8c531e8b447f1cd9127c8bc6ce8070c0e6/lib/kafka/produce_operation.rb#L170)) 

Would adding a new ddtrace method for this would be a valid approach to getting a topic-tagged span somewhere on the producer side?

Reply:
@cjford You're correct that there is additional instrumentation in ruby-kafka. When I implemented the current support in ddtrace I only included the instrumentation that used a block and therefore provided the timing for a span.

The `ack_message.producer` instrumentation that you identified is at the individual message level and would generate a lot of spans if you instrumented at that level, with no timing associated.

If you were going to build on that existing ruby-kafka instrumentation, then I'd probably explore using the `ack_message.producer` to add tags to the span (for `send_messages.producer`) that it is already nested within. 

Those tags could take the form a list of topics (because in general there will be multiple), or even a message count by topic.

Another alternative would be to add additional instrumentation to ruby-kafka first around the produce operation: https://github.com/zendesk/ruby-kafka/blob/93c45a8c531e8b447f1cd9127c8bc6ce8070c0e6/lib/kafka/produce_operation.rb#L125-L133
There are still multiple topics involved here, but it may be more straightforward to emit directly from this code the list of topics or the count of messages by topic.

Reply:
I think I'll go ahead and close this now, given that our project is likely migrating to a different kafka client in the near future, and more complex changes to ddtrace/ruby-kafka are getting a little out of my depth.

Thanks very much for your input. This should at least be a good reference if anyone else is looking to solve the same issue.